core/src/test/java/nl/nn/adapterframework/pipes/FilePipeTest.java

```diff
@@ -0,0 +1,92 @@
+package nl.nn.adapterframework.pipes;
+
+import nl.nn.adapterframework.core.IPipeLineSession;
+import nl.nn.adapterframework.core.PipeForward;
+import nl.nn.adapterframework.core.PipeLineSessionBase;
+import nl.nn.adapterframework.core.PipeRunException;
+import nl.nn.adapterframework.core.PipeRunResult;
+import nl.nn.adapterframework.parameters.Parameter;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.After;
+import org.junit.rules.TemporaryFolder;
+import org.mockito.Mock;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+
+import static org.junit.Assert.*;
+
+/**
+ * FilePipe Tester.
+ *
+ * @author <Sina Sen>
+ * @version 1.0
+ * @since <pre>Feb 28, 2020</pre>
+ */
+public class FilePipeTest extends PipeTestBase<FilePipe> {
+
+    @ClassRule
+    public static TemporaryFolder testFolderSource = new TemporaryFolder();
+
+    private static String sourceFolderPath;
+
+    @Mock
+    private IPipeLineSession session1 = new PipeLineSessionBase();
+
+    private byte[] var = "Some String you want".getBytes();
+    private ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(var);
+
+    @Override
+    public FilePipe createPipe() {
+        return new FilePipe();
+    }
+
+
+    @BeforeClass
+    public static void before() throws Exception {
+        testFolderSource.newFile("1.txt");
+        sourceFolderPath = testFolderSource.getRoot().getPath();
+
+    }
+
+
+    @Test
+    public void doTestSuccess() throws Exception {
+        Parameter p = new Parameter();
+        p.setSessionKey("key"); p.setName("p1"); p.setValue("15"); p.setType("int"); p.configure();
+        session1.put("key", p);
+        PipeForward fw = new PipeForward();
+        fw.setName("test"); pipe.registerForward(fw); pipe.addParameter(p);
+        pipe.setCharset("/"); pipe.setDirectory(sourceFolderPath);
+        pipe.setOutputType("stream"); pipe.setActions("read");
+        pipe.setFileName("1.txt"); pipe.setFileSource("filesystem");
+        pipe.setActions("create");
+        pipe.configure();
+        PipeRunResult res = pipe.doPipe(var, session1);
+        assertEquals(res.getPipeForward().getName(), "success");
+    }
+
+    @Test
+    public void doTestFailAsEncodingNotSupportedBase64() throws Exception {
+        exception.expect(PipeRunException.class);
+        exception.expectMessage("Pipe [FilePipe under test] msgId [null] Error while executing file action(s): (UnsupportedEncodingException) /");
+        Parameter p = new Parameter();
+        p.setSessionKey("key"); p.setName("p1"); p.setValue("15"); p.setType("int"); p.configure();
```

Comment:
Please put all the setters underneath each other for better readability.

Reply:
Thanks for the feedback, will do